opsdroid/connector/slack/connector.py

```diff
@@ -197,23 +194,30 @@ def _generate_base_data(self, event: opsdroid.events.Event) -> dict:
 
         return data
 
+    async def _retrieve_channels_from_slack(self, **kwargs):
```

Comment:
I'm curious why this is a new function, it seems very similar to `_get_channels`, even the docstrings are almost the same.

`Retrieve channels from the slack API` vs `Grab all the channels from the Slack API`

Reply:
The 3 instructions under `_retrieve_channels_from_slack` had to be placed twice under `_get_channels`, so thought of shortening them into one method, if this doesn't makes sense, I can move all back to `_get_channels`

Reply:
I think the added indirection here makes it harder to read.

Perhaps instead of calling `_retrieve_channels_from_slack` twice you could refactor so the `conversations_list` only happens once inside the loop.

```python
cursor = None

while True:
    channels = await self.slack_web_client.conversations_list(cursor=cursor, limit=self.channel_limit)
    self.known_channels.update({c["name"]: c for c in channels["channels"]})
    cursor = channels["response_metadata"].get("next_cursor")

    if not cursor:
        break
```

Reply:
Fair enough, I was calling it twice because I wasn't sure if `conversations_list` would take a `None` argument.  Will test it and modify accordingly

Reply:
I have simplified `_get_channels` method